allocator_test.go

```diff
@@ -184,6 +184,74 @@ func TestTagsWithNonviableNodes(t *testing.T) {
 	require.Nil(t, allocation)
 }
 
```

Comment:
General comment: these tests are getting a bit boiler-platey + unreadable. It'd be a good time to try the datadriven format we talked about.